src/components/WeaponComponent.vue

```diff
@@ -198,6 +227,85 @@ export default {
 		.camouflage {
 			user-select: none;
 
+			&.weapon-layout-grid > .inner {
+				flex-direction: column;
+				justify-content: center;
+
+				&.completed > .info {
+					opacity: 0.5;
+				}
+
+				img {
+					height: 100%;
+					object-fit: cover;
+					position: relative;
+					width: 100%;
+					z-index: 1;
+				}
+
+				.icon-component {
+					left: 50%;
+					opacity: 0;
+					position: absolute;
+					transform: translate(-50%, -50%);
+					transition: $transition;
+					top: 50%;
+					z-index: 2;
+				}
+
+				.info {
+					padding: 8px;
+
+					.name {
+						font-size: 14px;
+					}
+				}
+			}
+
+			&.weapon-layout-list > .inner {
+				$image-size: 30px;
+				background: $elevation-1-color;
+				flex-direction: row;
+				justify-content: flex-start;
+
+				&.completed > .info {
+					opacity: 0.5;
+				}
+
+				img {
+					height: 100%;
+					object-fit: cover;
+					position: relative;
+					width: $image-size;
+					z-index: 1;
+				}
+
+				.icon-component {
+					left: calc($image-size / 2);
+					opacity: 0;
+					position: absolute;
+					transform: translate(-50%, -50%);
+					transition: $transition;
+					top: 50%;
+					z-index: 2;
+				}
+
+				.info {
+					padding: 0 5px;
```

Comment:
Let's increase the padding a bit here and add y-axis padding for the requirements that span two rows (sniper rifles for example)
```suggestion
					padding: 4px 8px;
```